cmd/ndm_daemonset/controller/controller.go

```diff
@@ -213,6 +225,32 @@ func (c *Controller) setHostName() error {
 	} else {
 		c.NodeAttributes[HostNameKey] = hostName
 	}
+
+	labelList, err := getLabelList()
+	if err != nil {
+		return err
+	}
+	if labelList != "" {
+		labels := strings.Split(labelList, ",")
+		for _, label := range labels {
+			// check if the node-attribute label is present in the list
+			// If present then add all the node labels to the controller node attributes
+			if label == LabelTypeNode {
+				// Fill the blockdevice labels with all the topological labels of the node
+				// to which it is attached to
+				for key, value := range node.Labels {
+					if strings.Contains(key, "openebs.io") {
```

Comment:
Is this line required? If not specified, possibly the next line will still ensure that only one entry is present. Also, if there is an exlusion required, can this be more specific, what values are we skipping here? `hostname` ?

Reply:
Next line checks for the value emptiness but this one checks whether key contains any openebs related labels which is not required for node topolpgy or location details. Labels like: `openebs.io/nodename=node1-virtual-machine`,`topology.cstor.openebs.io/nodeName=node1-virtual-machine`,`topology.jiva.openebs.io/nodeName=node1-virtual-machine``. This are not required so this condition takes care of not adding any openebs specific labels.